src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer.Common/MEFProjectEngineFactories.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.AspNetCore.Razor.ProjectEngineHost;
+using Microsoft.CodeAnalysis.Razor;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Common;
+
+internal static class MEFProjectEngineFactories
+{
+    // The actual factories are defined in the ProjectEngineHost project, but we need to convert them
+    // to Lazy<,> to be MEF compatible.
+    public static readonly Lazy<IProjectEngineFactory, ICustomProjectEngineFactoryMetadata>[] Factories =
+        ProjectEngineFactories.Factories.Select(f => new Lazy<IProjectEngineFactory, ICustomProjectEngineFactoryMetadata>(() => f.Item1, f.Item2)).ToArray();
```

Comment:
The project engine factories were creating using MEF types, but MEF is not needed for hosting razor at all, so we don't want that. This wraps the non-MEF types.

The old file, now deleted but visible directly below this, wasn't using DI anyway, so no functional change really.